roles/deploy/tasks/main.yml

```diff
@@ -2,6 +2,10 @@
 - name: Initialize
   deploy_helper: path="{{ project_root }}" state=present
 
+- name: Update git remote URL
+  shell: "cd {{ project_source_path }} && git remote set-url origin {{ project_git_repo }} || :"
+  register: git_status
+
```

Comment:
@louim you always help me improve. Thank you. Consider this task:
```yaml
- name: Update git remote URL
  shell: "cd {{ project_source_path }} && git config --get remote.origin.url && git remote set-url origin {{ project_git_repo }} || :"
  register: original_remote
  changed_when: original_remote.stdout not in ['', project_git_repo]
```
On the surface, the task appears idempotent. It reports "changed" only if the new url differs from the old url. However, technically the `set-url` command always runs if the clone exists. If that is unacceptable, I could add an if-then into the shell command to conditionally run the `set-url` command. Or more likely, I could just let go of my drive to handle this in a single task and do three tasks like those you suggested.